controllers/core_clusteroperator_controller.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	openshiftconfigv1 "github.com/openshift/api/config/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logr "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	"github.com/openshift/platform-operators/internal/clusteroperator"
+)
+
+type CoreClusterOperatorReconciler struct {
+	client.Client
+	ReleaseVersion  string
+	SystemNamespace string
+}
+
+const (
+	coreCOName = "platform-operators-core"
+)
+
+//+kubebuilder:rbac:groups=platform.openshift.io,resources=platformoperators,verbs=list
+//+kubebuilder:rbac:groups=config.openshift.io,resources=clusteroperators,verbs=get;list;watch
+//+kubebuilder:rbac:groups=config.openshift.io,resources=clusteroperators/status,verbs=update;patch
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
+func (r *CoreClusterOperatorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := logr.FromContext(ctx)
+	log.Info("reconciling request", "req", req.NamespacedName)
+	defer log.Info("finished reconciling request", "req", req.NamespacedName)
+
+	coBuilder := clusteroperator.NewBuilder()
+	coWriter := clusteroperator.NewWriter(r.Client)
+
+	core := &openshiftconfigv1.ClusterOperator{}
+	if err := r.Get(ctx, req.NamespacedName, core); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return ctrl.Result{}, err
+		}
+
+		log.Info("core clusteroperator does not exist. recreating it...", "name", coreCOName)
+		core = &openshiftconfigv1.ClusterOperator{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: coreCOName,
+			},
+			Status: openshiftconfigv1.ClusterOperatorStatus{},
+		}
+		return ctrl.Result{}, r.Create(ctx, core)
+	}
```

Comment:
If the CO is in the payload, is there ever really a chance that our controller would need to re-create it? Is there a CVO doc about this?

If we do need to be able to re-create it ourselves, it looks like we need to add RBAC `create` permission for COs.

Reply:
https://github.com/openshift/enhancements/blob/master/dev-guide/cluster-version-operator/dev/clusteroperator.md#how-should-i-include-clusteroperator-custom-resource-in-manifests

> It remains a responsibility of the respective operator to properly update (or recreate if deleted) the ClusterOperator Custom Resource

I was reading through that documentation earlier today and spotted that documentation blob. Makes me question whether we should specify the create-only CVO annotation here if we're fully responsible for this resource outside of the initial rollout.

Reply:
> (or recreate if deleted)

Yeah, this seems to contrast with the fact that CVO would recreate it for us if we didn't use `create-only`. Maybe needs some clarification from @wking.

Reply:
Just anecdotal experience is that I can manually delete the ClusterOperator resource, and the CVO won't create a new one. I think the safest thing here is to update both ClusterOperator resources and specify the create-only annotation, and lean on our reconciliation logic to ensure this resource gets created by our controllers.

Reply:
This is also a gap in the current aggregate ClusterOperator controller implementation. It's a simply change to update the reconciliation logic to create the ClusterOperator if it doesn't already exist, and we can uniformly specify the `release.openshift.io/create-only: "true"` annotation.

Reply:
Talked about this in another PR and in slack. Expectation is that CVO#840 will remove the need for managing the ClusterOperator directly, so I removed this logic in recent force pushes/commits.